README.md

```diff
@@ -0,0 +1,180 @@
+# CoAP Client and Server for go
+
+Features supported:
+* CoAP over UDP [RFC 7252][coap].
+* CoAP over TCP/TLS [RFC 8232][coap-tcp]
+* request multiplexer
+
+Not yet implemented:
+* CoAP over DTLS
+
+Fork of https://github.com/dustin/go-coap
+
+[coap]: http://tools.ietf.org/html/rfc7252
+[coap-tcp]: https://tools.ietf.org/html/rfc8323
+
+## Samples
+
+### Simple
+
+#### Server UDP/TCP
+```go
+	// Server
+	// See /examples/simpler/server/main.go
+	func handleA(w coap.ResponseWriter, req coap.Message) {
+		log.Printf("Got message in handleA: path=%q: %#v from %v", req.Path(), req, w.RemoteAddr())
+		if req.IsConfirmable() {
+			res := w.NewMessage(coap.MessageParams{
+				Type:      coap.Acknowledgement,
+				Code:      coap.Content,
+				MessageID: req.MessageID(),
+				Token:     req.Token(),
+				Payload:   []byte("server: hello world!"),
+			})
+			res.SetOption(coap.ContentFormat, coap.TextPlain)
+
+			log.Printf("Transmitting from B %#v", res)
+			w.WriteMsg(res)
+		}
+	}
+
+	func main() {
+		mux := coap.NewServeMux()
+		mux.Handle("/a", coap.HandlerFunc(handleA))
+
+		log.Fatal(coap.ListenAndServe(":5688", "udp", mux))
+		
+		// for tcp
+		// log.Fatal(coap.ListenAndServe(":5688", "tcp", mux))
+
+		// fot tcp-tls
+		// log.Fatal(coap.ListenAndServeTLS(":5688", CertPEMBlock, KeyPEMBlock, mux))
+	}
+```
+#### Client
+```go
+	// Client
+	// See /examples/simpler/client/main.go
+	func main() {
+		co, err := coap.Dial("udp", "localhost:5688")
+		
+		// for tcp
+		// co, err := coap.Dial("tcp", "localhost:5688")
+		
+		// for tcp-tls
+		// co, err := coap.DialWithTLS("localhost:5688", &tls.Config{InsecureSkipVerify: true})
+
+		if err != nil {
+			log.Fatalf("Error dialing: %v", err)
+		}
+		defer co.Close()
```

Comment:
close the connection right away, defer for close might lead to running out of sockets.
also co.Close() error is ignored

Reply:
Yeah I understand. For example i removed this line because it is simple example